src/main/scala/com/ambiata/disorder/GenPlus.scala

```diff
@@ -1,9 +1,13 @@
 package com.ambiata.disorder
 
 import org.scalacheck._, Arbitrary._, Gen._
+import scala.collection.JavaConverters._
 
 object GenPlus {
 
+  def choose(from: Int, to: Int): Gen[Int] =
+    Gen.sized(n => Gen.choose(from, Math.max(from, Math.min(n, to))))
+
   // The Gen version of this function results in too many discarded tests
   def nonEmptyListOf[A](gen: => Gen[A]): Gen[List[A]] =
```

Comment:
I can't actually reproduce the normal version discarding values. I wonder whether it was something else. I'm tempted to remove this and `choose` but I guess there isn't much harm in leaving them for now.